src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -143,6 +143,35 @@ public IEnumerable<TagHelperDescriptor> Resolve(TagHelperDescriptorResolutionCon
             return descriptors;
         }
 
+        /// <summary>
+        /// This method returns a deep copy of a given <see cref="TagHelperDescriptor"/>.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/> to clone.</param>
+        /// <returns></returns>
+        public static TagHelperDescriptor CloneDescriptor(TagHelperDescriptor descriptor)
+        {
+            var cloneDescriptor = new TagHelperDescriptor
+            {
+                Prefix = descriptor.Prefix,
+                TagName = descriptor.TagName,
+                TypeName = descriptor.TypeName,
+                AssemblyName = descriptor.AssemblyName,
+                Attributes = descriptor.Attributes,
```

Comment:
Would the attributes need to be deep copied too (the same way the `PropertyBag` is)?